### ISSUE TITLE
Build fails when included in create-react-app

### ISSUE THREAD

**Issue by dan1elhughes** [ianstormtaylor/superstruct#86]
```
> npm run build

> <redacted>@0.1.0 build X:\github\<redacted>
> react-scripts build

Creating an optimized production build...
Failed to compile.

Failed to minify the code from this file:

        ./node_modules/superstruct/lib/index.es.js:7

Read more here: http://bit.ly/2tRViJ9

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ethereum-pos@0.1.0 build: `react-scripts build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ethereum-pos@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\<redacted>
```

Looking at the [link printed above](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#npm-run-build-fails-to-minify), it looks like `create-react-app` expects the node modules to be compiled ES5. Is there a way around this?

**Comment by ianstormtaylor** [comment]
@dan1elhughes why is it getting the `lib/index.es.js` file (which is `package.module`) instead of the `lib/index.js` file (which is `package.main`), do you know? There are two entry points, one is for CJS and one for ES modules.

**Comment by dan1elhughes** [comment]
Is it possible to publish `lib/index.js` pre-compiled to ES5, as part of the build process?

**Comment by ianstormtaylor** [comment]
Why is ES5 required? It should be the latest-ish ES* spec instead I'd think. Or rather, `create-react-app` shouldn't be failing when importing files that contain `class` declarations, since those are supported by every major browser already? It should compile them down if needed instead?

**Comment by dan1elhughes** [comment]
I've done some searching around and I'm struggling to figure out quite what's causing this. I've got [a workaround](https://github.com/dan1elhughes/ethereum-pos/commit/b58532ec71d8633851268a1614b21d47c147f68f) for now, but it's not pretty..

**Comment by jsleeuw** [comment]
I got around this with [react-app-rewire-babel-loader](https://github.com/dashed/react-app-rewire-babel-loader).

**Comment by dan1elhughes** [comment]
@jsleeuw That looks useful, thanks!
@ianstormtaylor I don't think Create React App is going to work out of the box with this unless it's ES5, as babel-compiling all of node modules would be very computationally heavy. I'm happy to investigate putting a build process in place to publish this as ES5, if that's a direction you're willing to go?

**Comment by ianstormtaylor** [comment]
I think the newer versions of create react app do babelize all node modules?

I’d be open to that PR. I’d just like to figure out what the root cause is before changing anything first.

**Comment by dan1elhughes** [comment]
Ah yes, looks like that is indeed the case in facebook/create-react-app#1125 . I'll update my project to latest CRA and see if that changes anything

**Comment by dan1elhughes** [comment]
Nope, turns out that feature is in the 2.x alpha versions, not the recently released 1.1.1 version. I'll look at creating that PR.

**Comment by dan1elhughes** [comment]
I'm no longer using this library so I'm going to close this issue. If this bug still exists, I'm sure it'll reappear in another issue as needed :-)
